tests/integration/test_repos_release.py

```diff
@@ -151,17 +158,19 @@ def test_edit(self):
         """Test the ability to edit an existing asset."""
         self.basic_login()
         cassette_name = self.cassette_name('edit')
-        with self.recorder.use_cassette(cassette_name,
-                                        preserve_exact_body_bytes=True):
-            repository = self.gh.repository('github3py', 'github3.py')
+        with self.recorder.use_cassette(cassette_name):
+            repository = self.gh.repository('itsmemattchung', 'github3.py')
             release = repository.create_release(
                 '0.8.0.pre', 'develop', '0.8.0 fake release with upload',
                 'To be deleted'
                 )
-            with open(__file__) as fd:
-                asset = release.upload_asset(
-                    'text/plain', 'test_repos_release.py', fd.read()
-                    )
+        with self.recorder.use_cassette(cassette_name,
```

Comment:
hey @sigmavirus24 tests are passing but I ran into a snag and think there may be a better way to address this.

First, some of these tests have multiple requests which are either **body** and **json-body**.  To handle this with the same cassette, I use two *with* statements, with **body** and **json-body** as matchers for betamax - respectively.  Not sure if there's a way to check the cassette for `['uri','method','body']` OR `['uri, 'method', 'json-body']` and reduce it down to one with statement. If so, the second question can be disregarded (below)

Second, by referencing the same cassette twice in the test, I had to use `record='new_episodes'` the first time when creating the cassette if not the second **with** statement would fail since it can't find a matching request. After the cassette is recorded, I removed `record='new_episodes'` since I want future tests to fail if the test changes. I think this could cause confusion in the future if someone wants to re-record the test.